fms/utils/serialization.py

```diff
@@ -56,6 +66,97 @@ def list_sources(architecture: str):
     return list(__adapters[architecture].keys())
 
 
+def _legacy_attn_unfused_to_fused_weight_conversion(
+    name: str, orig_sd: Mapping
+) -> Optional[Tuple[str, torch.Tensor]]:
+    """
+    function which converts unfused fms weights to fused fms weights in the case the model was using the older unfused
+    weights (version 0.0.5)
+
+    Args:
+        name: str
+            current name to convert
+        orig_sd: Mapping
+            a mapping from a name to a param, in most cases a grouping of all names to params in a module.
+
+    Returns:
+    Optional[Tuple[str, torch.Tensor]]
+        if query/key/value all exist in the given state dict, a tuple of the new fused name as well as weights will be
+        returned from this function
+        otherwise this function will return None signifying that no preprocessing needed to be done for this name param
+    """
+    # if we find query/key/value, this means the weights are unfused and therefore need to be fused
+    if "attn.query" in name or "attn.key" in name or "attn.value" in name:
+        weight_type = name.split(".")[-1]
```

Comment:
yes I believe its weight vs bias

Reply:
might help to have a comment here just so it's more obvious to the reader... as I was reading I was getting confused thinking it would be one of key/query/value and then had to re-read again after the below